src/TmsIntegrationExample.Example/TmsIntegrationExampleProcessor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// <copyright file="MockProcessor.cs" company="Engaged Technologies">
+//      Copyright (c) Engaged Technologies. All rights reserved.
+//  </copyright>
+// <summary>
+//     implementation class for processing Mock orders/shipments
+// </summary>
+
+namespace TmsIntegrationExample.Example
+{
+    using Microsoft.Extensions.Options;
+    using System.IO;
+    using System.Reflection;
+    using System.Threading.Tasks;
+    using TmsIntegrationExample.Example.Transformers;
+    using TmsIntegrationExample.Services.MockApi;
+    using TmsIntegrationExample.Services.TmsGatewayApi;
+    using TmsIntegrationExample.Services.TmsGatewayApi.Configuration;
+
+    public class TmsIntegrationExampleProcessor : ITmsIntegrationExampleProcessor
+    {
+        private readonly ITmsGatewayClient tmsGatewayClient;
+        private readonly IMockClient mockClient;
+        private readonly TmsGatewayConfiguration tmsGatewayConfiguration;
+
+        public TmsIntegrationExampleProcessor(
+            ITmsGatewayClient tmsGatewayClient, 
+            IMockClient mockClient,
+            IOptions<TmsGatewayConfiguration> tmsGatewayConfiguration)
+        {
+            this.tmsGatewayConfiguration = tmsGatewayConfiguration.Value;
+            this.tmsGatewayClient = tmsGatewayClient;
+            this.mockClient = mockClient;
+        }
+
+        public async Task<int> ProcessAsync()
+        {
+            var ordersResult = await mockClient.GetOrdersAsync(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), @"MockApi\Orders.json"));
+
+            var numberProcessed = 0;
+
+            if (ordersResult != null)
+            {
+                foreach (var order in ordersResult)
+                {
+                    var sourceLocation = await mockClient.GetAddressByAddressIdAsync(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), @"MockApi\SourceLocation.json"));
+                    var shipTo = new Services.MockApi.Model.Address();
+                    if (order.ShipToId != null)
+                    {
+                        shipTo = await mockClient.GetAddressByAddressIdAsync(Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), @"MockApi\ShipTo.json"));
+                    }
+
+                    var shipmentPost = new MockOrderToShipment().Transform(order, shipTo, sourceLocation, this.tmsGatewayConfiguration.AccountNumber);
+
+                    await tmsGatewayClient.PostShipmentAsync(shipmentPost);
+
+                    numberProcessed++;
```

Comment:
This might be a little too picky but the result of this will always be 0, the total number of orders, or an exception will be thrown somewhere that will make it so this method doesn't return anyway. As it stands, this method will never return something between 0 and the total number of orders since there is no condition that will not increment.

Reply:
Unless there was something specific we were trying to illustrate here, we can just return `orders.Count()`. I'm sure the exact collection name is different but I'm not looking at the code right now other than this little bit github shows in the convo view.

Reply:
code changed to return the count. please review if it requires any aditional modification.